django/core/management/commands/shell.py

```diff
@@ -111,10 +115,35 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_and_report_namespace(self, verbosity):
+        namespace = self.get_namespace()
+
+        if verbosity >= 1:
+            self.stdout.write(
+                f"{len(namespace)} objects imported automatically",
+                self.style.SUCCESS,
+            )
+
+        return namespace
+
+    def get_namespace(self):
+        apps_models = apps.get_models()
+        apps_models_modules = [
+            (app.models_module, app.label) for app in apps.get_app_configs()
+        ]
+        namespace = {}
+        for model in reversed(apps_models):
+            if model.__module__:
+                namespace[model.__name__] = model
+        for app in apps_models_modules:
+            app_models_module, app_label = app
```

Comment:
Use a multi-target loop:

```suggestion
        for app_models_module, app_label in apps_models_modules:
```